R/create_project.R

```diff
@@ -0,0 +1,120 @@
+"dwapi-r
+Copyright 2018 data.world, Inc.
+
+Licensed under the Apache License, Version 2.0 (the \"License\");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an \"AS IS\" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+This product includes software developed at data.world, Inc.
+https://data.world"
+
+#' Create a new project.
+#' @param owner_id data.world user name of the project owner.
+#' @param create_project_req Request object of type \code{\link{project_create_request}}.
+#' @return Object of type \code{\link{create_project_response}}.
+#' @examples
+#' request <- dwapi::project_create_request(
+#'   title='testproject', visibility = 'OPEN',
+#'   objective = 'Test project by R-SDK', tags = c('rsdk', 'sdk', 'arr'),
+#'   license = 'Public Domain')
+#'
+#' request <- dwapi::add_file(request = request, name = 'file4.csv',
+#'   url = 'https://data.world/file4.csv')
+#'
+#' \dontrun{
+#' dwapi::create_project(create_project_req = request,
+#'   owner_id = 'user')
+#' }
+#' @export
+create_project <- function(owner_id, create_project_req) {
+  create_or_update_project(owner_id, create_project_req, httr::POST)
+}
+
+#' Create a new project with a specific ID, replacing a project with that ID
+#' if it exists.
+#' @param owner_id data.world user name of the project owner.
+#' @param project_id identifier to use for the new project, or identifier of
+#' existing project to replace with the newly created project.
+#' @param create_project_req Request object of type \code{\link{project_create_request}}.
```

Comment:
I noticed that we have `dataset_replace_request` and `insight_replace_request`. They are identical to their `*_create_request` counterparts yet, we've decided to use different types to represent them. For consistency, should we have `project_replace_request` here?

Reply:
Yes, and now we do!